src/connectDialog.cpp

```diff
@@ -86,10 +91,54 @@ void ConnectDialog::btnClicked()
 		QString new_uri = URIstringParser(ui->hostname->text());
 		Q_EMIT newContext(new_uri);
 	} else {
+		if (enableDemo) {
+			// iio-emu found in system
+			QString program = "iio-emu";
+			QStringList arguments;
+			arguments.append(ui->demoDevicesComboBox->currentText());
+			process->setProgram(program);
+			process->setArguments(arguments);
+			process->start();
+			auto started = process->waitForStarted();
+			if (!started) {
+				// retry to start the process
+				program = "iio-emu/iio-emu";
+				process->setProgram(program);
+				process->start();
+				started = process->waitForStarted();
+				if (!started) {
+					qDebug() << "Process failed to start";
+				}
+			}
+			if (started) {
+				qDebug() << "Process started successfully";
+				QThread::msleep(100);
```

Comment:
Why do we need this sleep here? It currently blocks the main thread for `100ms`

Reply:
This is a simple way to ensure that the server is prepare to listen to the clients, otherwise the validateInput function cannot create a context.
There is also the possibility to send a message when the server is ready.